tests/data/semparse/worlds/nlvr_world_test.py

```diff
@@ -90,3 +90,98 @@ def test_logical_form_with_box_filter_returns_correct_action_sequence(self):
                                    "<b,e> -> ['lambda x', e]", 'e -> [<#1,e>, o]', '<#1,e> -> count',
                                    'o -> [<o,o>, o]', '<o,o> -> blue', 'o -> [<b,o>, b]', '<b,o> -> object_in_box',
                                    'b -> [<#1,#1>, b]', '<#1,#1> -> var', 'b -> x', 'e -> 0', 'e -> 1']
+
+    def test_get_logical_form_with_real_logical_forms(self):
+        nlvr_world = self.worlds[0]
+        logical_form = ("(assert_greater_equals \
+                          (count (filter_equals all_boxes (lambda x (count (blue (object_in_box (var x))))) 0)) \
+                          1)")
+        parsed_logical_form = nlvr_world.parse_logical_form(logical_form)
+        action_sequence = nlvr_world.get_action_sequence(parsed_logical_form)
+        reconstructed_logical_form = nlvr_world.get_logical_form(action_sequence)
+        parsed_reconstructed_logical_form = nlvr_world.parse_logical_form(reconstructed_logical_form)
+        # It makes more sense to compare parsed logical forms instead of actual logical forms.
```

Comment:
Why?

Reply:
Added to comment:
".. because there can be slight differences between the actual logical form strings, like extra set of parentheses or spaces, which neither the type inference logic nor the executor cares about. So the test shouldn't either."